pkg/sync/grpc/grpc_sync.go

```diff
@@ -171,14 +195,61 @@ func (g *Sync) handleFlagSync(stream syncv1grpc.FlagSyncService_SyncFlagsClient,
 	}
 }
 
-// URLToGRPCTarget is a helper to derive GRPC target from a provided URL
+// buildTransportCredentials is a helper to build grpc credentials.TransportCredentials based on source and cert path
+func buildTransportCredentials(source string, certPath string) (credentials.TransportCredentials, error) {
+	if strings.Contains(source, Prefix) {
+		return insecure.NewCredentials(), nil
+	}
+
+	if !strings.Contains(source, PrefixSecure) {
+		return nil, fmt.Errorf("invalid source. grpc source must must contain prefix %s or %s", Prefix, PrefixSecure)
+	}
+
+	var tCred credentials.TransportCredentials
+
+	if certPath == "" {
+		// Rely on CA certs provided from system
+		tCred = credentials.NewTLS(&tls.Config{MinVersion: tlsVersion})
+	} else {
+		// Rely on provided certificate
+		certBytes, err := os.ReadFile(certPath)
+		if err != nil {
+			return nil, err
+		}
+
+		cp := x509.NewCertPool()
+		if !cp.AppendCertsFromPEM(certBytes) {
+			return nil, fmt.Errorf("invalid certificate provided at path: %s", certPath)
+		}
+
+		tCred = credentials.NewTLS(&tls.Config{
+			MinVersion: tlsVersion,
+			RootCAs:    cp,
+		})
+	}
```

Comment:
This might be a bit better as a ternary assignment with the provided credentials coming from an extracted function (so that `tCred` is never a zero-value).

Reply:
Good suggestion, I will work on improving the logic flow.

Reply:
Resolved with the latest commit :)